src/travel-details-screens/utils.ts

```diff
@@ -160,3 +161,14 @@ export function canSellCollabTicket(tripPattern: TripPattern) {
 function someLegsAreByTrain(tripPattern: TripPattern): boolean {
   return tripPattern.legs.some((leg) => leg.mode === Mode.Rail);
 }
+
+/**
+ * Convert a `Mode` value to `TransportMode`. If the mode isn't a valid
+ * transport mode, return 'unknown'.
+ */
+export function modeToTransportMode(mode: Mode): TransportMode {
+  if (Object.values(TransportMode as any).includes(mode)) {
+    return mode as any as TransportMode;
+  }
+  return TransportMode.Unknown;
+}
```

Comment:
Also possible to use `AnyMode` (and `AnySubMode`) in the component params, making this mapping perhaps not necessary, or at least done closer to where it is necessary to have it.